static/web.js

```diff
@@ -69,11 +69,23 @@ function format(result, session, version) {
     });
 }
 
+// URL safe alphabet, no padding: https://tools.ietf.org/html/rfc4648#section-5
+// encodeURIComponent is due to https://developer.mozilla.org/en-US/docs/Web/API/WindowBase64.btoa#Unicode_Strings
+function b64_encode(str) {
+    return btoa(encodeURIComponent(str)).replace(/\+/g, '-').replace(/\//g, '_').replace(/\=+$/, '');
```

Comment:
Are the two layers of encoding necessary? Could `encodeURIComponent` be eschewed in favor of just base64 encoding?